logstash-core/src/main/java/org/logstash/instrument/metrics/counter/LongCounter.java

```diff
@@ -0,0 +1,78 @@
+package org.logstash.instrument.metrics.counter;
+
+
+import org.logstash.instrument.metrics.AbstractMetric;
+import org.logstash.instrument.metrics.MetricType;
+
+import java.util.List;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * A {@link CounterMetric} that is backed by a {@link Long} type.
+ */
+public class LongCounter extends AbstractMetric<Long> implements CounterMetric<Long> {
+
+    private final LongAdder longAdder;
+
+    /**
+     * Constructor - initial value set to zero
+     *
+     * @param nameSpace The namespace for this metric
+     * @param key       The key <i>(with in the namespace)</i> for this metric
+     */
+    public LongCounter(List<String> nameSpace, String key) {
+        this(nameSpace, key, 0l);
+    }
+
+    /**
+     * Constructor
+     *
+     * @param nameSpace    The namespace for this metric
+     * @param key          The key <i>(with in the namespace)</i> for this metric
+     * @param initialValue The initial value for this {@link CounterMetric}
+     */
+    public LongCounter(List<String> nameSpace, String key, long initialValue) {
```

Comment:
Removed in the latest round of changes: https://github.com/elastic/logstash/pull/7595/commits/6cb5902f000c2b92004a5e593e1e44476b7c9c6a